lms/djangoapps/certificates/migrations/0009_certificategenerationcoursesetting_language_self_generation.py

```diff
@@ -0,0 +1,40 @@
+# -*- coding: utf-8 -*-
+from __future__ import unicode_literals
+
+from django.apps import apps
+from django.db import migrations, models
+from django.db.models import F
+
+def copy_field(apps, schema_editor):
+    CertificateGenerationCourseSetting = apps.get_model('certificates', 'CertificateGenerationCourseSetting')
+    CertificateGenerationCourseSetting.objects.all().update(self_generation_enabled=F('enabled'))
+
+class Migration(migrations.Migration):
+    """
+    Adds new field 'language_specific_templates_enabled'.
+    Also adds field 'self_generation_enabled' which is a 
+    replacement for 'enabled'
+    Lastly, copies data from 'enabled' to 'self_generation_enabled'
+    """
+    dependencies = [
+        ('certificates', '0008_schema__remove_badges'),
+    ]
+
+    operations = [
+        migrations.AddField(
+            model_name='certificategenerationcoursesetting',
+            name='language_specific_templates_enabled',
+            field=models.BooleanField(default=False, help_text="Render translated certificates rather than using the platform's default language. Available translations are controlled by the certificate template."),
+        ),
+        migrations.AddField(
+            model_name='certificategenerationcoursesetting',
+            name='self_generation_enabled',
+            field=models.BooleanField(default=False, help_text='Allow students to generate their own certificates for the course. Enabling this does NOT affect usage of the management command used for batch certificate generation.'),
+        ),
+        migrations.AlterField(
+            model_name='certificategenerationcoursesetting',
+            name='enabled',
+            field=models.BooleanField(default=False, help_text='DEPRECATED, please use self_generation_enabled instead.'),
+        ),
+        migrations.RunPython(copy_field),
```

Comment:
@feanil this last operation in this migration copies data from the old enabled field into the newly named field.

Reply:
This should stop self generation from just turning off for all courses.

Reply:
@Mjloturco what happens if this step fails?

Reply:
Add a reverse function, which should be [`RunSQL.noop`](https://docs.djangoproject.com/en/1.11/ref/migration-operations/#django.db.migrations.operations.RunPython.noop).

Reply:
@feanil since there are so few records in this table, what are the downsides to just renaming the field via migration vs creating a new field and copying over the old data?

Reply:
https://engineering.edx.org/django-migration-donts-f4588fd11b64